server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/RestSubsystemAdd.java

```diff
@@ -58,13 +58,12 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
 
       String virtualServer = config.hasDefined(ModelKeys.VIRTUAL_SERVER) ? config.get(ModelKeys.VIRTUAL_SERVER).asString() : DEFAULT_VIRTUAL_SERVER;
```

Comment:
You need to setHostname() on the NettyJaxrsServer based on the virtual server name.